### PR TITLE
Move unix socket to v2 input API

### PR DESCRIPTION
**Pull request by urso**
- Refactoring

## What does this PR do?

Move the "unix" input to input v2 API

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Author's Checklist

- [ ]

## How to test this PR locally

run `test_unix.py` system test.

## Related issues

- Relates #15324

### CODE REVIEW

File: filebeat/input/default-inputs/inputs.go [hunk 1]
```diff
@@ -32,5 +33,7 @@ func Init(info beat.Info, log *logp.Logger, components beater.StateStore) []v2.P
 }
 
 func genericInputs() []v2.Plugin {
-	return []v2.Plugin{}
+	return []v2.Plugin{
+		unix.Plugin(),
```
Comment (urso) [inline filebeat/input/default-inputs/inputs.go:37]: Adding unix socket to genericInputs that are supported on all platforms.

Newer windows versions do support AF_UNIX, but only streaming unix sockets AFAIK

File: filebeat/input/unix/input.go [hunk 2]
```diff
@@ -18,118 +18,97 @@
...
-			Timestamp: time.Now(),
-			Fields: common.MapStr{
-				"message": string(data),
-			},
-		})
-	}
+	return newServer(config)
+}
 
+func newServer(config config) (*server, error) {
```
Comment (urso) [inline filebeat/input/unix/input.go:61]: the constructor (in this case `newServer`) should not depend on `common.Config`. The `configure` function is introduced to unpack the config and pass it to the typed constructor.

File: filebeat/input/default-inputs/inputs.go (hunk 1 above)
Comment (kvch) [reply filebeat/input/default-inputs/inputs.go:37]: What does that mean from support standpoint? Filebeat supports older Windowses, e.g. Windows Server 2012, which cannot handle AF_UNIX based on my quick internet research. How are we informing users about the limitation of this input?

Comment (urso) [reply filebeat/input/default-inputs/inputs.go:37]: We already link the unix input into Filebeat as is. So it is no real change. Removing it from windows build would be a breaking change.

On older windows versions the input will fail to create the listener. That's all

**Pull request merged**